numpy-stubs/__init__.pyi

```diff
@@ -919,3 +920,69 @@ def argsort(
     kind: Optional[_SortKind] = ...,
     order: Union[None, str, Sequence[str]] = ...,
 ) -> ndarray: ...
+@overload
+def argmax(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: None = ...,
+    out: Optional[ndarray] = ...,
+) -> integer: ...
+@overload
+def argmax(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: int = ...,
+    out: Optional[ndarray] = ...,
+) -> Union[integer, ndarray]: ...
+@overload
+def argmin(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: None = ...,
+    out: Optional[ndarray] = ...,
+) -> integer: ...
+@overload
+def argmin(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: int = ...,
+    out: Optional[ndarray] = ...,
+) -> Union[integer, ndarray]: ...
+@overload
+def searchsorted(
+    a: Union[Sequence[ArrayLike], ndarray],
+    v: _Scalar,
+    side: _Side = ...,
+    sorter: Union[None, Sequence[_IntOrBool], ndarray] = ...,  # 1D int array
```

Comment:
I don't love the complexity things like this add to each signature versus having some uniform "array like but not a scalar" type that we can use everywhere. This is less detailed for now, but we can reach for this level of detail when making `ndarray`/`ArrayLike` generic; i.e. we could have something like `_NonScalarArrayLike[T]` and keep the code more readable/maintainable. Right now each signature ends up being its own little puzzle.

But, I haven't complained about this on previous PRs, so maybe now is not the time to start objecting.

Reply:
I do like the `_NonScalarArrayLike[T]` suggestion. 

A question though: do you feel it would be better to implement such a change in this pull request? Or shall i create a new pull request afterwards, addressing the issue from both https://github.com/numpy/numpy-stubs/pull/71 and https://github.com/numpy/numpy-stubs/pull/67?